src/main/java/dinkplugin/notifiers/DeathNotifier.java

```diff
@@ -33,57 +52,94 @@ public void onActorDeath(ActorDeath actor) {
-                pker = other;
-                break;
-            }
-        }
-        ItemContainer inventory = plugin.getClient().getItemContainer(InventoryID.INVENTORY);
-        ItemContainer equipment = plugin.getClient().getItemContainer(InventoryID.EQUIPMENT);
+        Actor pker = identifyPker();
 
-        assert inventory != null;
-        assert equipment != null;
-        List<Pair<Item, Long>> itemsByPrice = Stream.concat(Arrays.stream(inventory.getItems()), Arrays.stream(equipment.getItems()))
-            .map(item -> Pair.of(item, (long) (plugin.getItemManager().getItemPrice(item.getId())) * (long) (item.getQuantity())))
-            .sorted((item1, item2) -> Math.toIntExact(item2.getRight() - item1.getRight()))
-            .collect(Collectors.toList());
+        List<Pair<Item, Long>> itemsByPrice = getPricedItems();
 
-        int keepCount = 3;
-        if (localPlayer.getSkullIcon() != null) {
+        int keepCount;
+        if (localPlayer.getSkullIcon() == null)
+            keepCount = 3;
+        else
             keepCount = 0;
-        }
-        if (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM)) {
-            keepCount += 1;
-        }
-        Long losePrice = itemsByPrice.stream()
+        if (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM))
+            keepCount++;
+
+        long losePrice = itemsByPrice.stream()
             .skip(keepCount)
             .map(Pair::getRight)
             .reduce(Long::sum)
             .orElse(0L);
 
-        String template = plugin.getConfig().deathNotifyMessage();
-        if (pker != null && plugin.getConfig().deathNotifPvpEnabled()) {
+        boolean pvpNotif = pker != null && plugin.getConfig().deathNotifPvpEnabled();
+        String template;
+        if (pvpNotif)
             template = plugin.getConfig().deathNotifPvpMessage();
-        }
+        else
+            template = plugin.getConfig().deathNotifyMessage();
         String notifyMessage = template
             .replace("%USERNAME%", Utils.getPlayerName(plugin.getClient()))
-            .replace("%VALUELOST%", losePrice.toString());
-        if (pker != null && plugin.getConfig().deathNotifPvpEnabled()) {
-            // player name hopefully isn't null
-            notifyMessage = notifyMessage
-                .replace("%PKER%", Objects.requireNonNull(pker.getName()));
+            .replace("%VALUELOST%", String.valueOf(losePrice));
+        if (pvpNotif) {
+            notifyMessage = notifyMessage.replace("%PKER%", String.valueOf(pker.getName()));
         }
 
+        List<NotificationBody.Embed> lostItemEmbeds = itemsByPrice.stream()
+            .skip(keepCount)
+            .map(Pair::getLeft)
+            .mapToInt(Item::getId)
+            .distinct()
+            .limit(3)
+            .mapToObj(Utils::getItemImageUrl)
+            .map(NotificationBody.UrlEmbed::new)
+            .map(NotificationBody.Embed::new)
+            .collect(Collectors.toList());
```

Comment:
Could you add these to the DeathNotificationData preferably with their names and whatnot?

Reply:
Just the top 3 or all lost items?

also which pieces of info?
- [ ] item id
- [ ] item name
- [ ] total quantity of item being lost (across stacks)
- [ ] image url

Reply:
i think having item id, name, quantity of the top 3 should be good enough, i think we have all the info to create a `SerializedItemStack`.

Reply:
This has been implemented but ought to be tested (notifier complexity has increased)